File: CppPractice/Project1/Project1/Leetcode171.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

using namespace std;


class Solution {
public:
    int titleToNumber(string columnTitle) {
        int ans = 0;
        for (int i = columnTitle.size() - 1; i >= 0; i--)
        {
            ans += pow(26, columnTitle.length() - 1 - i) * (columnTitle[i] - 'A' + 1);
        }
        return ans;

    }
};

File: Pythonleetcode/Leetcode171.py
class Solution(object):
    def titleToNumber(self, s):
        ans=0
        value_base=ord("A")
        for i in range(len(s)-1,-1,-1):
            ans=(26**(len(s)-1-i))*(ord(s[i])-value_base+1)+ans
            print(i, s[i],ans)

        return ans




print(Solution().titleToNumber("AB"))

